frontend/src/ui/wallet/shared/confirm-intention-view.js

```diff
@@ -0,0 +1,33 @@
+import React, {useEffect, useState} from 'react'
+import {AccountAddress} from '@stellar-expert/ui-framework'
+import {fetchAssetPrices} from "../../../state/ledger-data/asset-price"
+
+export function ConfirmIntentionView({transfer}) {
+    const selfTransfer = transfer.source === transfer.destination
+    const [estimatedText, setEstimatedText] = useState('')
+    const accountName = window.predefinedAccountDisplayNames[transfer.source]
+
+    useEffect(() => {
+        (async () => {
+            try {
+                const prices = await fetchAssetPrices(transfer.network, [transfer.asset[0]])
+                const estimatedValue = Object.values(prices)[0]
+                const estimatedPrice = transfer.amount[0] * estimatedValue
+                if (estimatedPrice) setEstimatedText(`(~${estimatedPrice.toFixed(2)}$)`)
+            } catch (e) {
+                console.log(e)
+            }
+        })()
```

Comment:
This should be completely rewritten.
- In such cases promise chained functions look much more organic (`fetchAssetPrices(transfer.network, [transfer.asset[0]]).then(prices=>...).catch(e=>console.error(e))`)
- Never use `console.log()` for displaying errors. `console.error()` was invented for this specific case
- **Do not** write `if` condition and `then` branch in the same line. **Never**.
- Never use `.toFixed` for formatting prices or other unknown numerical values. Price can be 10000000 or 0.0000001. In both cases `.toFixed()` will return ugly looking price. Methods `.toFixed()` and `.toPrecision()` are **prohibited** in our codebase unless you are writing a low-level helper method for formatting. We have `formatPrice()`, `formatWithPrecision()`, `formatWithAutoPrecision()`, `formatWithAbbreviation()` helpers in the `@stellar-expert/formatter` lib for this exact purpose.